x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/UnaryScalarFunction.java

```diff
@@ -7,23 +7,125 @@
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Ceil;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Cos;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Cosh;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Floor;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Log10;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Signum;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Sin;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Sinh;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Sqrt;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Tan;
+import org.elasticsearch.xpack.esql.expression.function.scalar.math.Tanh;
+import org.elasticsearch.xpack.esql.expression.function.scalar.spatial.StX;
+import org.elasticsearch.xpack.esql.expression.function.scalar.spatial.StY;
+import org.elasticsearch.xpack.esql.expression.function.scalar.string.LTrim;
+import org.elasticsearch.xpack.esql.expression.function.scalar.string.Length;
+import org.elasticsearch.xpack.esql.expression.function.scalar.string.RTrim;
+import org.elasticsearch.xpack.esql.expression.function.scalar.string.Trim;
+import org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.Neg;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamInput;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamOutput;
 
+import java.io.IOException;
 import java.util.Arrays;
+import java.util.List;
 
 import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isNumeric;
 
 public abstract class UnaryScalarFunction extends EsqlScalarFunction {
+    public static List<NamedWriteableRegistry.Entry> getNamedWriteables() {
+        return List.of(
+            Abs.ENTRY,
+            Acos.ENTRY,
+            Asin.ENTRY,
+            Atan.ENTRY,
+            Cbrt.ENTRY,
+            Ceil.ENTRY,
+            Cos.ENTRY,
+            Cosh.ENTRY,
+            Floor.ENTRY,
+            FromBase64.ENTRY,
+            Length.ENTRY,
+            Log10.ENTRY,
+            LTrim.ENTRY,
+            Neg.ENTRY,
+            RTrim.ENTRY,
+            Signum.ENTRY,
+            Sin.ENTRY,
+            Sinh.ENTRY,
+            Sqrt.ENTRY,
+            StX.ENTRY,
+            StY.ENTRY,
+            Tan.ENTRY,
+            Tanh.ENTRY,
+            ToBase64.ENTRY,
+            ToBoolean.ENTRY,
+            ToCartesianPoint.ENTRY,
+            ToDatetime.ENTRY,
+            ToDegrees.ENTRY,
+            ToDouble.ENTRY,
+            ToGeoShape.ENTRY,
+            ToCartesianShape.ENTRY,
+            ToGeoPoint.ENTRY,
+            ToIP.ENTRY,
+            ToInteger.ENTRY,
+            ToLong.ENTRY,
+            ToRadians.ENTRY,
+            ToString.ENTRY,
+            ToUnsignedLong.ENTRY,
+            ToVersion.ENTRY,
+            Trim.ENTRY
```

Comment:
Should we add a note to the "how to write a scalar function" docs to mention you need to add it here?

Reply:
:+1: